src/scraper/http.cpp

```diff
@@ -141,6 +231,162 @@ std::string Http::GetEtag(
         _log(logattribute::INFO, "curl_http_header", "Captured ETag for project url <urlfile=" + url + ", ETag=" + etag + ">");
 }
 
+std::string Http::GetLatestVersionResponse(const std::string& url)
+{
+    std::string buffer;
+    std::string header;
+
+    struct curl_slist* headers = NULL;
+    headers = curl_slist_append(headers, "Accept: */*");
+    headers = curl_slist_append(headers, "User-Agent: curl/7.63.0");
+
+    ScopedCurl curl = GetContext();
+    curl_easy_setopt(curl.get(), CURLOPT_URL, url.c_str());
+    curl_easy_setopt(curl.get(), CURLOPT_WRITEFUNCTION, curl_write_string);
+    curl_easy_setopt(curl.get(), CURLOPT_WRITEDATA, &buffer);
+    curl_easy_setopt(curl.get(), CURLOPT_HEADERDATA, &header);
+    curl_easy_setopt(curl.get(), CURLOPT_HTTPHEADER, headers);
+
+    CURLcode res = curl_easy_perform(curl.get());
+
+    if (res > 0)
+        throw std::runtime_error(tfm::format("Failed to get ETag for URL %s: %s", url, curl_easy_strerror(res)));
+
+    curl_slist_free_all(headers);
+
+    // Validate HTTP return code.
+    long response_code;
+    curl_easy_getinfo(curl.get(), CURLINFO_RESPONSE_CODE, &response_code);
+    EvaluateResponse(response_code, url);
+
+    // Return the Http response
+    return buffer;
+}
+
+void Http::DownloadSnapshot(
+        const std::string &url,
+        const std::string &destination)
+{
+    ScopedFile fp(fsbridge::fopen(destination.c_str(), "wb"), &fclose);
```

Comment:
i did exactly what u had changed above for Http::Download. does this mean urs needs change as well?

Reply:
not trying to be confusing ;p

Reply:
You're right, looks like I might have missed some of those bits with the search and replace. I didn't notice because it's all scraper-specific functionality that doesn't get much exercise on Windows. I'll go back and fix those in another PR.

Reply:
r u sure i can always fix it within this since i'm in that area of code? save u a rebase?